theano/tensor/basic.py

```diff
@@ -5888,41 +5888,89 @@ def all(x, axis=None, keepdims=False):
 class Diagonal(Op):
     """Return specified diagonals.
 
+    Usage: Diagonal(offset, axis1, axis2)(x)
+
+    `x` has to be a tensor with 2 or more dimensions.
+
     Parameters
     ----------
-    x
+    offset : int
+        Indicates which diagonal to extract, `0` as main diagonal, positive as
+        diagonals in the upper triangle, and vice versa.
+
+    axis1, axis2 : int
+        Indicates on which dimensions to extract the diagonals.
+
+    x: symbolic tensor
         A tensor variable with x.ndim >= 2.
 
     Returns
     -------
-    vector
+    vector : symbolic vector
         A vector representing the diagonal elements.
 
     """
-    __props__ = ("offset", "axis1", "axis2")
+    __props__ = ("offset", "axis1", "axis2", "view")
+    default_offset = 0
+    default_axis1 = 0
+    default_axis2 = 1
 
-    def __init__(self, offset=0, axis1=0, axis2=1):
-        if numpy_diagonal_return_view:
+    def __init__(self, offset=0, axis1=0, axis2=1, view=False):
+        self.view = view
```

Comment:
You should not set view = True if numpy_diagonal_return_view is False.

Actually, to make it more transparent I would do something like this:

```python
def __init__(self, offset=0, axis1=0, axis2=1, view=False):
    self.view = view
    if self.view and not numpy_diagonal_return_view:
        warnings.warn("View will forced to False. Diagonal property view is "
                      "set to True but numpy version %s and prior versions of "
                      "numpy.diagonal() do not return a view. Update "
                      "numpy to use Diagonal(view=True)" % 
                      numpy.version.version)
        self.view = False
    if self.view:
        self.view_map = {0: [0]}
```